datahub/views.py

```diff
@@ -2390,6 +2389,80 @@ def requested_datasets(self, request, *args, **kwargs):
     #     serializer = self.get_serializer(page, many=True).exclude(is_temp = True)
     #     return self.get_paginated_response(serializer.data)
 
+    @action(detail=True, methods=["get"])
+    def get_dashboard_chart_data(self, request, pk, *args, **kwargs):
+        try:
+            dataset_file_object = DatasetV2File.objects.get(id=pk)
+            dataset_file = str(dataset_file_object.standardised_file)
+            try:
+                if dataset_file.endswith(".xlsx") or dataset_file.endswith(".xls"):
+                    df = pd.read_excel(os.path.join(settings.DATASET_FILES_URL, dataset_file))
+                elif dataset_file.endswith(".csv"):
+                    df = pd.read_csv(os.path.join(settings.DATASET_FILES_URL, dataset_file))
+
+                else:
+                    return Response(
+                        "Unsupported file please use .xls or .csv.",
+                        status=status.HTTP_400_BAD_REQUEST,
+                    )
+                obj = {
+                    "total_no_of_records": len(df),
+                    "male_count": (df['Gender'] == 1).sum(),
+                    "female_count": (df['Gender'] == 0).sum(),
+                    "constituencies": (df['Constituency']).nunique(),
+                    "counties": (df['County']).nunique(),
+                    "sub_counties": (df['Sub County']).nunique(),
+                    "farming_practices": {
+                        "crop_production": (df['Crop Production']).sum(),
+                        "livestock_production": (df['Livestock Production']).sum(),
+                    },
```

Comment:
First get the available columns and do the manipulations on only available columns  , Try to lower all the column names and do the process.

Existing code will work for only positive  flow